superset/utils/core.py

```diff
@@ -815,12 +815,15 @@ def __exit__(  # pylint: disable=redefined-outer-name,redefined-builtin
 
 def pessimistic_connection_handling(some_engine: Engine) -> None:
     @event.listens_for(some_engine, "engine_connect")
-    def ping_connection(connection: Connection, branch: bool) -> None:
+    def on_connect(connection: Connection, branch: bool) -> None:
         if branch:
             # 'branch' refers to a sub-connection of a connection,
             # we don't want to bother pinging on these.
             return
 
+        if connection.dialect.name == "sqlite":
```

Comment:
See [here](https://docs.sqlalchemy.org/en/20/dialects/sqlite.html#foreign-key-support) for why this is needed. I do wonder if we should drop support for SQLite given it's not supported in production. We already skip a bunch of tests, i.e., 

```python
if backend() == "sqlite":
    return
```

possibly because of how the foreign key constraints have no impact on the underlying table.